packages/docs/src/pages/components/index.mdx

```diff
@@ -95,6 +97,14 @@ The `sx` prop on Theme UI components can be used *without* the `jsx` pragma comm
 
 </Note>
 
+<Note>
+
+If you’re using TypeScript, Theme UI merges prop types with the props of components passed in the `as` prop.
```

Comment:
> Theme UI merges prop types with the props of components passed in the `as` prop.

We do?

I remember I just surrendered with Box for `@types/theme-ui__components`, because supporting `as` prop on Box hurt the performance so much I couldn't pass DefinitelyTyped tests (and VSCode was starting to behave funny).

Box is a styled component, so it was already quite heavy on TypeScript, and adding `as` was the last straw that breaks the back.

We only support `as` in TypeScript on `Field`, where it's crucial.

![image](https://user-images.githubusercontent.com/15332326/99941872-08f7a800-2d6f-11eb-9848-5a46f269811e.png)

Reply:
We could support `as` if we got rid of styled-system, and emotion/styled (so `as` instead of `withComponent`).

We could deprecate `withComponent` first, though.

Reply:
Yep you're totally right, I misread a sentence in https://github.com/system-ui/theme-ui/issues/1124#issuecomment-672719902

Will take a look at this later today!

Reply:
Thanks!